rln/src/public.rs

```diff
@@ -550,6 +553,64 @@ mod test {
         assert_eq!(root_delete, root_empty);
     }
 
+    #[test]
+    // We test leaf setting with a custom index, to enable batch updates to the root
+    // Uses `set_leaves_from` to set leaves in a batch, from index `start_index`
+    fn test_leaf_setting_with_index() {
+        let tree_height = TEST_TREE_HEIGHT;
+        let no_of_leaves = 256;
+        let start_index = 5;
+        let default_leaf = Fr::from(0);
+
+        // We generate a vector of random leaves
+        let mut leaves: Vec<Fr> = Vec::new();
+        let mut rng = thread_rng();
+        for _ in 0..no_of_leaves {
+            leaves.push(Fr::rand(&mut rng));
+        }
+
+        // We create a new tree
+        let input_buffer = Cursor::new(TEST_RESOURCES_FOLDER);
+        let mut rln = RLN::new(tree_height, input_buffer);
+
+        // We add leaves in a batch into the tree
+        let mut buffer = Cursor::new(vec_fr_to_bytes_le(&leaves));
+        rln.set_leaves_from(start_index, &mut buffer).unwrap();
+
+        // We check if number of leaves set is consistent
+        assert_eq!(rln.tree.leaves_set(), no_of_leaves + start_index);
+
+        // We get the root of the tree obtained adding leaves in batch
+        let mut buffer = Cursor::new(Vec::<u8>::new());
+        rln.get_root(&mut buffer).unwrap();
+        let (root_batch, _) = bytes_le_to_fr(&buffer.into_inner());
+
+        // We reset the tree to default
+        rln.set_tree(tree_height).unwrap();
+
+        // add `start_index` empty leaves
+        for _ in 0..start_index {
+            let mut buffer = Cursor::new(fr_to_bytes_le(&default_leaf));
```

Comment:
Addressed in [bb0027e](https://github.com/vacp2p/zerokit/pull/63/commits/bb0027ed09d7625aefabcd3e4b4c46faa7f7c828)

Reply:
> `set_leaves` is currently not exposed in rln-wasm. If you want you could add it, but otherwise it can be done in a future PR when it is required!

https://github.com/vacp2p/zerokit/issues/64 tracking here :)